ingestion/src/metadata/ingestion/source/metadata/alationsink/client.py

```diff
@@ -0,0 +1,207 @@
+from metadata.utils.helpers import clean_uri
+from metadata.utils.logger import utils_logger
+
+logger = utils_logger()
+
+
+class AlationSinkAuthenticationProvider(AuthenticationProvider):
+    """
+    Module to Handle Alation Auth
+    """
+
+    def __init__(self, config: AlationSinkConnection):
+        self.config = config
+        self.service_connection = self.config
+        client_config = ClientConfig(
+            base_url=clean_uri(config.hostPort),
+            api_version="integration/v1",
+            auth_token=lambda: ("no_token", 0),
+            auth_header="Authorization",
+            allow_redirects=True,
+        )
+        self.client = REST(client_config)
+        self.generated_auth_token = None
+        self.expiry = None
+        super().__init__()
+
+    @classmethod
+    def create(cls, config: AlationSinkConnection):
+        return cls(config)
+
+    def auth_token(self) -> None:
+        """
+        Generate the auth token
+        """
+        if isinstance(self.config.authType, ApiAccessTokenAuth):
+            self.generated_auth_token = (
+                self.config.authType.accessToken.get_secret_value()
+            )
+            self.expiry = 0
+        else:
+            self._get_access_token_from_basic_auth()
+
+    def get_access_token(self):
+        """
+        Return the generated access token
+        """
+        self.auth_token()
+        return self.generated_auth_token, self.expiry
+
+    def _get_access_token_from_basic_auth(self):
+        """
+        Get the access token for basic auth type
+        """
+        # Get the refresh token
+        refresh_token_data = {
+            "username": self.config.authType.username,
+            "password": self.config.authType.password.get_secret_value(),
+            "name": self.config.projectName,
+        }
+        refresh_token_response = self.client.post(
+            "/createRefreshToken/", json.dumps(refresh_token_data)
+        )
+
+        # Get the access token
+        access_token_data = {
+            "refresh_token": refresh_token_response["refresh_token"],
+            "user_id": refresh_token_response["user_id"],
+        }
+        access_token_response = self.client.post(
+            "/createAPIAccessToken/", json.dumps(access_token_data)
+        )
+
+        self.generated_auth_token = access_token_response["api_access_token"]
+        self.expiry = 0
+
+
+class AlationSinkClient:
+    """
+    Client to interact with Alation apis
+    """
+
+    def __init__(self, config: AlationSinkConnection):
+        self.config = config
+        self._auth_provider = AlationSinkAuthenticationProvider.create(config)
+        client_config: ClientConfig = ClientConfig(
+            base_url=clean_uri(config.hostPort),
+            auth_header="TOKEN",
+            api_version="integration",
+            auth_token=self._auth_provider.get_access_token,
+            auth_token_mode="",
+            verify=False,
```

Comment:
should this be hardcoded as `False`? We might need to allow some tuning of the certs here

Reply:
Got it I'll add ssl methods